lazyClient.go

```diff
@@ -27,111 +27,91 @@ func NewMSSelect(c io.ReadWriteCloser, proto string) Multistream {
 // perform any protocol selection. If you are using a MultistreamMuxer, use
 // NewMSSelect.
 func NewMultistream(c io.ReadWriteCloser, proto string) Multistream {
-	return &lazyConn{
+	return &lazyClientConn{
 		protos: []string{proto},
 		con:    c,
 	}
 }
 
-type lazyConn struct {
-	rhandshake bool // only accessed by 'Read' should not call read async
+type lazyClientConn struct {
```

Comment:
can we get a doc comment saying what the purpose of `lazyClientConn` is exactly?